File: v1/src/main/java/com/api/v1/person/dtos/PersonRegistrationRequestDto.java
package com.api.v1.person.dtos;

import com.api.v1.person.annotations.SSN;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

import java.time.LocalDate;

public record PersonRegistrationRequestDto(
        @NotBlank String firstName,
        String middleName,
        @NotBlank String lastName,
        @SSN String ssn,
        LocalDate birthDate,
        @NotBlank @Email String email,
        @NotBlank String address,
        @NotBlank @Size(min = 10, max = 10) String phoneNumber,
        @NotBlank @Size(min = 1) String gender
) {
}

File: kotlin/v3/src/main/kotlin/com/api/v3/persons/dtos/PersonRegistrationRequestDto.kt
package com.api.v3.persons.dtos

import java.time.LocalDate

data class PersonRegistrationRequestDto(
    val firstName: String,
    val middleName: String?,
    val lastName: String,
    val ssn: String,
    val email: String,
    val birthDate: LocalDate,
    val address: String,
    val gender: String,
    val phoneNumber: String
)
